faucet/src/bin/server.rs

```diff
@@ -0,0 +1,69 @@
+use axum::{
+    http::StatusCode,
+    response::IntoResponse,
+    routing::{get, post},
+    Extension, Json, Router,
+};
+use std::net::SocketAddr;
+use sui::{
+    sui_config_dir,
+    wallet_commands::{WalletCommands, WalletContext},
+    SUI_WALLET_CONFIG,
+};
+use sui_faucet::{FaucetRequest, Service, SimpleFaucet};
+use tracing::info;
+
+#[tokio::main]
+async fn main() -> Result<(), anyhow::Error> {
+    // initialize tracing
+    tracing_subscriber::fmt::init();
+
+    let context = create_wallet_context().await?;
+
+    let app = Router::new()
+        .route("/", get(health))
+        .route("/gas", post(request_gas))
+        .layer(Extension(Service::new(
+            SimpleFaucet::new(context).await.unwrap(),
+        )));
```

Comment:
Instead of wrapping this in the `Service` type, can we just create 1 SimpleFaucet and then pass it in here as an extension? Then that 1 SimpleFaucet will be cloned and provided for each request. Unless is there some reason a new Facuet needs to be created on each request?